libvast/src/system/index.cpp

```diff
@@ -158,15 +157,18 @@ caf::error index_state::load_from_disk() {
           VAST_WARNING(self, "could not mmap partition at", partition_path);
           continue;
         }
-        auto partition = fbs::as_versioned_flatbuffer<fbs::Partition>(
-          as_bytes(chunk), fbs::Version::v1);
-        if (!partition) {
-          VAST_WARNING(self, "coult not parse", partition_path,
-                       "as partition:", partition.error());
+        auto partition = fbs::GetPartition(chunk->data());
+        if (partition->versioned_partition_type()
+            != fbs::PartitionUnion::v1_Partition) {
+          // TODO: We could also support v0 partitions here by creating the
+          // missing meta index on-the-fly.
```

Comment:
Does it make sense to throw out the v0 and make v1 the v0 instead, since they're both happening for the same release? (Unless this does not make the cut)